process/smartContract/processorV2/processV2.go

```diff
@@ -2325,12 +2329,19 @@ func (sc *scProcessor) preprocessOutTransferToSCR(
 	result.GasLimit = outputTransfer.GasLimit
 	result.CallType = outputTransfer.CallType
 	setOriginalTxHash(result, txHash, tx)
-	if result.Value.Cmp(zero) > 0 {
-		result.OriginalSender = tx.GetSndAddr()
-	}
+	result.OriginalSender = GetOriginalSenderForTx(tx)
 	return result
 }
 
+func GetOriginalSenderForTx(tx data.TransactionHandler) []byte {
```

Comment:
add comment to public function.

Reply:
It would be harder but we could have implemented this function inside data.TransactionHandler - but that is a bigger job. can leave as it is. only add the comment